design/airgap-existing-cluster.md

```diff
@@ -0,0 +1,110 @@
+# Support for installing KOTS Admin Console in Airgapped Clusters
+
+## Goals
+
+- Install kotsadm to an existing cluster where the cluster does not have internet access
+- Install kotsadm to an existing cluster from a workstation where the workstation does not have internet access at the time of installation
+- Support different credentials for push / pull (workstation / cluster) to the local registry
+- Support temporary/dynamic and persistent/static credentials for pushing kotsadm images to the registry
```

Comment:
I know it would be great to support a wide variety of credential methods, but I think if we wanted to we could cut a little scope here and have any registry push situation require passing a username and password via environment or CLI flags. Support for other fanciers auth methods could come later.

Reply:
To be clear, we're not really doing anything too special to support this other than asking for the push credentials separately from the pull credentials, and only using the push credentials once. If it required more work, then we could see about cutting it from scope, but as is, it was a natural part of the design anyway, so no additional work, yet still beneficial.